### PR TITLE
avoid LTO issues

### PR DESCRIPTION
**Pull request by neheb**
Place structs inside the places where used.

### CODE REVIEW

**Comment by neheb** [comment]
Fixes: https://github.com/Exiv2/exiv2/issues/2680

File: src/pentaxmn_int.cpp [hunk 1]
```diff
@@ -1231,25 +1231,22 @@ static std::ostream& resolveLensType(std::ostream& os, const Value& value, const
...
-    {0x031d, resolveLensType}, {0x031f, resolveLensType},  {0x0329, resolveLensType},  {0x032c, resolveLens0x32c},
-    {0x032e, resolveLensType}, {0x0334, resolveLensType},  {0x03ff, resolveLens0x3ff}, {0x041a, resolveLensType},
-    {0x042d, resolveLensType}, {0x08ff, resolveLens0x8ff},
-};
-
 //! A lens id and a pretty-print function for special treatment of the id.
 static std::ostream& printLensType(std::ostream& os, const Value& value, const ExifData* metadata) {
+  //! List of lens ids which require special treatment using resolveLensType
+  static constexpr struct LensIdFct {
+    long id_;       //!< Lens id
```
Comment (kevinbackhouse) [inline src/pentaxmn_int.cpp:1238]:
```suggestion
    uint32_t id_;       //!< Lens id
```

File: src/pentaxmn_int.cpp [hunk 2]
```diff
@@ -1231,25 +1231,22 @@ static std::ostream& resolveLensType(std::ostream& os, const Value& value, const
...
+    PrintFct fct_;  //!< Pretty-print function
+    //! Comparison operator for find template
+    bool operator==(long id) const {
```
Comment (kevinbackhouse) [inline src/pentaxmn_int.cpp:1241]:
```suggestion
    bool operator==(uint32_t id) const {
```

**Review by kevinbackhouse — COMMENTED** [review]
The difference between the two structs is that one has a `uint32_t` field and the other `long`. I'd like to remove all occurrences of `long` anyway, so I suggest we use `uint32_t` in both. With that change, this struct could potential be moved to a shared header file.

**Comment by neheb** [comment]
Done.

**Review by 1div0 — APPROVED** [review]
LG2M

**Comment by neheb** [comment]
@mergifyio backport 0.28.x

**Pull request merged**